src/components/internal/OverlayTrigger.tsx

```diff
@@ -110,3 +111,8 @@ export function isIconButton(
 ): trigger is IconButtonTriggerProps {
   return trigger && typeof trigger === "object" && "icon" in trigger;
 }
+
+// adding typeguard for items in props to conditionally render inside ButtonMenu
+export function isItemsProps(props: ButtonMenuProps): props is ButtonMenuProps {
```

Comment:
**suggestion:** When we do these typeguards we should be specify what `prop` could be. Something like:
```
function isItemsProps(props: ButtonMenuProps | SomeOtherProps): props is ButtonMenuProps {
```

Reply:
got it because its not only using `WithMenuItemsProps` but also uses `ButtonMenuBaseProps`